python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -439,14 +439,18 @@ def _create_node(self, node_config, tags, count):
 
                         # The correct value is technically
                         # {"code": "0", "Message": "pending"}
-                        state_reason = instance.state_reason or {"Message": "pending"}
+                        state_reason = "pending"
+                        if instance.state_reason:
+                            state_reason = (
+                                instance.state_reason["Message"] or state_reason
```

Comment:
in python 3.9, even when instance.state_reason is not None, it's 'Message' entry can still be None